src/sds.c

```diff
@@ -60,6 +60,24 @@ static inline int sdsHdrSize(char type) {
     return 0;
 }
 
+/* Returns size of sdsHdr by checking pointer alignment (expects 8 byte
+ * alignment). Optimizes retriving sdsHdr size by not refering to sds data */
+static inline int sdsHdrSizeOptim(char* s) {
+    switch((uintptr_t)s&7) {
+        case sizeof(struct sdshdr5)&7:
```

Comment:
Maybe some macro for this?
SDS_MOD8(T) (sizeof(struct sdshdr##T)&7)

and  instead of case sizeof(struct sdshdr5)&7: 

you could call:

case SDS_MOD8(5):

SDS_MOD8 or SDS_ALIGN I believe you could pick better name for this